.github/drafts/17478/17478_Active_Directory_Draft.md

```diff
@@ -0,0 +1,110 @@
+| **Azure Active Directory Configuration** | **OIDC configuration field in Appsmith** |
+| ---------------------------------------- | ---------------------------------------- |
+| authorization\_endpoint                  | Authorization URL                        |
+| token\_endpoint                          | Token URL                                |
+| userinfo\_endpoint                       | User Info URL                            |
+| jwks\_uri                                | JWK Set URL                              |
+
+![Appsmith - OIDC Setup](../../../../../.gitbook/assets/Appsmith-Admin-Settings-Authentication-OIDC-Setup.png)
+
+### Configure Scopes 
+
+The scope defines the OpenID Connect (OIDC) scopes that allow you to authorize the access of user details ( after a user is successfully authenticated) like name, email, profile picture, and more. Each scope maps to a set of user attributes and returns its value. You'll see the Scope field below the **JSON Web Key Set**:
+
+![Configure Scope(s) at Appsmith](./as_oidc_offline.png) <!-- updated for 17478 -->
+
+#### What does Appsmith need as part of Scopes? <!-- updated for 17478 -->
+
+Appsmith needs **openid** as a mandatory scope. It's also highly recommended to use the **offline_access** scope to avoid errors related to expired access tokens and excessive re-login requests.
+
+{% hint style="info" %}
+Enabling the `offline_access` scope enables your app to receive refresh tokens that extend the duration that your users have access to their resources. To read more, see the [Active Directory documentation](https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-permissions-and-consent#offline_access).
+{% endhint %}
+
+You can add more scopes if you wish, provided that they're available via Active Directory.
+
+### Configure Scopes in Active Directory
+
+{% embed url="https://youtu.be/D1qA11NKbNY" %} <!-- replaced for 17478 -->
+Configuring scopes in Active Directory
+{% endembed %}
+
+To configure scopes/permissions on Active Directory, follow the steps below:
+
+1. Go to API permissions on the left panel.
+2. Click on Add a permission button.
+3. Under Request API permissions, Select Microsoft Graph and click on Delegated permissions.
+4. Select the following permissions:
+   1. Under **OpenID** permissions-
+      1. email
+      2. offline_access
+      3. openid
+      4. Profile
+   2. Under **User** permissions-
+      1. User.Read
+5. Once complete, these scopes will get added to the Active Directory application.
+6. Click on **Grant admin consent for Default Directory** on the top of the table, next to Add a permission button.
+
+### Configure Username Attributes for Active Directory
+
+The username attributes define the attributes used as usernames for authentication. You can add the attribute to this field that you consider for logging.
+
+![Appsmith Username Attribute](<../../../../../.gitbook/assets/OIDC \_ Active Directory \_ Username Attribute \_ sub.png>)
```

Comment:
The image here doesn't have `offline_access` in the scope which might confuse users.